EventProc/src/HcalDigiProducer.cxx

```diff
@@ -128,76 +145,151 @@ namespace ldmx {
+                }
+                if (cur_layer % 2 == 1){ // odd layers, horizontal
+                    cur_ypos = (super_strip_width * (float(cur_strip)+0.5)) - total_width/2.; 
+                    cur_xpos = hcalXpos[detIDraw] + random_->Gaus(0.,150.); 
+                }
+                if (cur_xpos > total_width/2.) cur_xpos = total_width/2.;
+                if (cur_xpos < -1.*total_width/2.) cur_xpos = -1.*total_width/2.;
+                if (cur_ypos > total_width/2.) cur_ypos = total_width/2.;
+                if (cur_ypos < -1.*total_width/2.) cur_ypos = -1.*total_width/2.;
+                // std::cout << "super_strip_width = " << super_strip_width << "\t total_super_strips = " << total_super_strips << "\t total_width = " << total_width << std::endl;
+            }
+
+            if( hcalLayerPEs[detIDraw] >= readoutThreshold_ ){ // > or >= ?
+                
                 HcalHit *hit = (HcalHit*) (hits_->ConstructedAt(ihit));
                 
                 hit->setID(detIDraw);
                 hit->setPE(hcalLayerPEs[detIDraw]);
+                hit->setMinPE(hcalLayerMinPEs[detIDraw]);
                 hit->setAmplitude(hcalLayerPEs[detIDraw]);
                 hit->setEnergy(energy);
                 hit->setTime(hcaldetIDTime[detIDraw]);
-                hit->setXpos(hcalXpos[detIDraw]);
-                hit->setYpos(hcalYpos[detIDraw]);
+                // hit->setXpos(hcalXpos[detIDraw]);
+                // hit->setYpos(hcalYpos[detIDraw]);
+                hit->setXpos(cur_xpos); // quantized and smeared positions
+                hit->setYpos(cur_ypos); // quantized and smeared positions
                 hit->setZpos(hcalZpos[detIDraw]);
                 hit->setNoise(false);
                 ihit++;
                 
             }
+
+            if (verbose_) {
+                detID_->setRawValue(detIDraw);
+                detID_->unpack();
+                int layer = detID_->getFieldValue("layer");
+                int subsection = detID_->getFieldValue("section");
+                int strip = detID_->getFieldValue("strip");
+
+                std::cout << "detID: " << detIDraw << std::endl;
+                std::cout << "Layer: " << layer << std::endl;
+                std::cout << "Subsection: " << subsection << std::endl;
+                std::cout << "Strip: " << strip << std::endl;
+                std::cout << "Edep: " << hcaldetIDEdep[detIDraw] << std::endl;
+                std::cout << "numPEs: " << hcalLayerPEs[detIDraw] << std::endl;
+                std::cout << "time: " << hcaldetIDTime[detIDraw] << std::endl;
+                std::cout << "z: " << hcalZpos[detIDraw] << std::endl;
+                std::cout << "Layer: " << layer << "\t Strip: " << strip << "\t X: " << hcalXpos[detIDraw] <<  "\t Y: " << hcalYpos[detIDraw] <<  "\t Z: " << hcalZpos[detIDraw] << std::endl;
+            }        // end verbose            
         } 
         
```

Comment:
If I understand well, the noise is simulated per strip, i.e. we have either noise is the two readouts of a given strip (one value might be zero), or no noise in both readouts. Is that the correct model, or should we distribute noise hits in readouts instead of strips?

Reply:
Noise is simulated per readout (or SiPM).  And then the total PE and the minPE are saved (such that the maxPE could also be calculated)